### ISSUE TITLE
Global `fetch` type is getting modified by shiki

### ISSUE THREAD

**Issue by Ayc0** [shikijs/shiki#454]
Having this code in shiki:

```ts
declare global {
  function fetch(url: string): Promise<Response>
}
```

also impacts the global `fetch` when using shiki, and this type isn't compatible with the native one provided by TS

**Comment by orta** [comment]
Yeah, makes sense, I think the answer is to probably switch these

https://github.com/shikijs/shiki/blob/99a9a3dcd2f2504d36b0672968fb05be53fd73cb/packages/shiki/src/loader.ts#L7-L19

to be inline `declare xyc` in this file, which should make them not afffect the global scopes

**Comment by Ayc0** [comment]
@orta I can try to draft a PR if you want

**Comment by orta** [comment]
Give it a shot!

**Comment by Ayc0** [comment]
@orta the PR is ready: https://github.com/shikijs/shiki/pull/457

**Comment by octref** [comment]
Fixed by #457. Thanks for your contribution!

**Comment by klippx** [comment]
No, it is even worse now. Please reopen the issue! 

Before I added `nextra` as a dependency to my project I had  `await result.text()` returning typed as `string` automatically by typescript lib dom:

```ts
interface Body {
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/body) */
    readonly body: ReadableStream<Uint8Array> | null;
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/bodyUsed) */
    readonly bodyUsed: boolean;
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/arrayBuffer) */
    arrayBuffer(): Promise<ArrayBuffer>;
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/blob) */
    blob(): Promise<Blob>;
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/formData) */
    formData(): Promise<FormData>;
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/json) */
    json(): Promise<any>;
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/text) */
    text(): Promise<string>;
}
```

However, now this library got sucked in to the mix, it has ruined the `Response` typings:

```
declare global {
    interface Response {
        json(): Promise<any>;
        text(): Promise<any>;
    }
}
```

Can you please get rid of this completely? You are polluting everyone else's type safety, this is extremely bad practice to patch the global namespace like this.
